extensions/gc/tasks/GarbageCollectorTask.js

```diff
@@ -267,7 +267,11 @@ class GarbageCollectorTask extends BackbeatTask {
             },
         ], err => {
             if (err) {
-                // if error occurs, do not commit offset
+                // if error occurs, do not commit offset unless the error is ObjNotFound
+                // because it means the object has been deleted by other means and we don't need to retry
+                if (err.code === 'ObjNotFound') {
+                    return done(err, { committable: true });
```

Comment:
Not sure we should return an error here ?

Reply:
The fact that the error is committable doesnt change the fact that there is an error, i think we should still return it, to log it etc..